src/nimlsp.nim

```diff
@@ -382,28 +416,32 @@ proc main(ins: Stream | AsyncFile, outs: Stream | AsyncFile) {.multisync.} =
              
           of "textDocument/rename":
             message.textDocumentRequest(RenameParams, renameRequest):
-              debugLog "Running equivalent of: use ", uriToPath(fileuri), ";", filestash, ":",
-                rawLine + 1, ":",
-                openFiles[fileuri].fingerTable[rawLine].utf16to8(rawChar)
-              let suggestions = getNimsuggest(fileuri).use(uriToPath(fileuri), dirtyfile = filestash,
+              debugLog "Running equivalent of: use ", uriToPath(fileuri), ";",
+                        filestash, ":",
+                        rawLine + 1, ":",
+                        openFiles[fileuri].fingerTable[rawLine].utf16to8(rawChar)
+              let suggestions = getNimsuggest(fileuri).use(
+                uriToPath(fileuri),
+                dirtyfile = filestash,
                 rawLine + 1,
                 openFiles[fileuri].fingerTable[rawLine].utf16to8(rawChar)
               )
               debugLog "Found suggestions: ",
-                suggestions[0..(if suggestions.len > 10: 10 else: suggestions.high)],
-                (if suggestions.len > 10: " and " & $(suggestions.len-10) & " more" else: "")
+                suggestions[0..<min(suggestions.len, 10)],
+                if suggestions.len > 10: &" and {suggestions.len-10} more" else: ""
               var resp: JsonNode
               if suggestions.len == 0:
                 resp = newJNull()
               else:
                 var textEdits = newJObject()
                 for suggestion in suggestions:
-                  if not textEdits.hasKey("file://" & pathToUri(suggestion.filepath)):
-                    textEdits["file://" & pathToUri(suggestion.filepath)] = newJArray()
-                  textEdits["file://" & pathToUri(suggestion.filepath)].add create(TextEdit,
-                    create(Range,
+                  var k = "file://" & pathToUri(suggestion.filepath)
```

Comment:
This should be a `let` and be named something better than `k` perhaps simply just `filepath`.